### PR TITLE
Fix compilation on M1 macOS

### PR DESCRIPTION
**Pull request by Absolucy**
Homebrew on M1 uses `/opt/homebrew` instead of `/usr/local`

In addition, I added support for Procursus, which uses apt and installs in `/opt/procursus` on both x86_64 and M1.

### CODE REVIEW

File: osx_environment.sh [hunk 1]
```diff
@@ -29,18 +29,15 @@ if [ ! $BARRIER_BUILD_ENV ]; then
...
         export BARRIER_BUILD_BREW=1
-        export CMAKE_PREFIX_PATH="$QT_PATH:$CMAKE_PREFIX_PATH"
-        export LD_LIBRARY_PATH="$OPENSSL_PATH/lib:$LD_LIBRARY_PATH"
-        export CPATH="$OPENSSL_PATH/include:$CPATH"
-        export PKG_CONFIG_PATH="$OPENSSL_PATH/lib/pkgconfig:$PKG_CONFIG_PATH"
-
+        export CMAKE_PREFIX_PATH="/opt/procursus:$QT_PATH:$CMAKE_PREFIX_PATH"
+        export LD_LIBRARY_PATH="/opt/procursus/lib:$LD_LIBRARY_PATH"
+        export CPATH="/opt/procursus/include:$CPATH"
+        export PKG_CONFIG_PATH="/opt/procursus/lib/pkgconfig:$PKG_CONFIG_PATH"
```
Comment (yan12125) [inline osx_environment.sh:40]: I know little about Procursus, but it seems strange to include Pocursus paths when Homebrew is detected.

**Comment by OKNoah** [comment]
Running file on all 3 executables shows they are not yet universal

```
file Barrier.app/Contents/MacOS/barrier 
Contents/MacOS/barrier: Mach-O 64-bit executable x86_64
```

Keep up the great effort

**Comment by yan12125** [comment]
Looks like this PR only attempts to fix building but not to create universal binaries. Also, Homebrew does not seems to support universal binaries, either. https://github.com/Homebrew/brew/issues/10307

**Comment by yan12125** [comment]
There are now conflicting files as part of this PR is covered by https://github.com/debauchee/barrier/pull/1207

**Comment by fdelucchijr** [comment]
> Running file on all 3 executables shows they are not yet universal
> 
> ```
> file Barrier.app/Contents/MacOS/barrier 
> Contents/MacOS/barrier: Mach-O 64-bit executable x86_64
> ```

I get `Mach-O 64-bit executable arm64` for all 3 binaries, so, I think is ready 🤔. 
The problem at this point look more as a signing error, look at #960
Cheers!

**Comment by fdelucchijr** [comment]
> > Running file on all 3 executables shows they are not yet universal
> > ```
> > file Barrier.app/Contents/MacOS/barrier 
> > Contents/MacOS/barrier: Mach-O 64-bit executable x86_64
> > ```
> 
> I get `Mach-O 64-bit executable arm64` for all 3 binaries, so, I think is ready 🤔.
> The problem at this point look more as a signing error, look at #960
> Cheers!

For anyone facing this, use [this comment command](https://github.com/debauchee/barrier/issues/960#issuecomment-907273209).

**Comment by SiliconSausage** [comment]
> > > Running file on all 3 executables shows they are not yet universal
> > > ```
> > > file Barrier.app/Contents/MacOS/barrier 
> > > Contents/MacOS/barrier: Mach-O 64-bit executable x86_64
> > > ```
> > 
> > 
> > I get `Mach-O 64-bit executable arm64` for all 3 binaries, so, I think is ready 🤔.
> > The problem at this point look more as a signing error, look at #960
> > Cheers!
> 
> For anyone facing this, use [this comment command](https://github.com/debauchee/barrier/issues/960#issuecomment-907273209).
> 
> Also, I think this PR should be merged, Barrier running native Apple Arch in M1 MBA here!

Could you share a zip of the app please? I haven't got it to work. Thanks.

**Comment by fdelucchijr** [comment]
> 
> 
> > > > Running file on all 3 executables shows they are not yet universal
> > > > ```
> > > > file Barrier.app/Contents/MacOS/barrier 
> > > > Contents/MacOS/barrier: Mach-O 64-bit executable x86_64
> > > > ```
> > > 
> > > 
> > > I get `Mach-O 64-bit executable arm64` for all 3 binaries, so, I think is ready 🤔.
> > > The problem at this point look more as a signing error, look at #960
> > > Cheers!
> > 
> > 
> > For anyone facing this, use [this comment command](https://github.com/debauchee/barrier/issues/960#issuecomment-907273209).
> > Also, I think this PR should be merged, Barrier running native Apple Arch in M1 MBA here!
> 
> Could you share a zip of the app please? I haven't got it to work. Thanks.

Of course! Here is my build.
https://we.tl/t-jHlDF34qcV

**Comment by SiliconSausage** [comment]
> > > > > Running file on all 3 executables shows they are not yet universal
> > > > > ```
> > > > > file Barrier.app/Contents/MacOS/barrier 
> > > > > Contents/MacOS/barrier: Mach-O 64-bit executable x86_64
> > > > > ```
> > > > 
> > > > 
> > > > I get `Mach-O 64-bit executable arm64` for all 3 binaries, so, I think is ready 🤔.
> > > > The problem at this point look more as a signing error, look at #960
> > > > Cheers!
> > > 
> > > 
> > > For anyone facing this, use [this comment command](https://github.com/debauchee/barrier/issues/960#issuecomment-907273209).
> > > Also, I think this PR should be merged, Barrier running native Apple Arch in M1 MBA here!
> > 
> > 
> > Could you share a zip of the app please? I haven't got it to work. Thanks.
> 
> Of course! Here is my build. https://we.tl/t-jHlDF34qcV

Thank you SO much🙏🏾

 I had to run CHMOD-X on the each of the three executables in the MacOS fodler in the app package but now it's working perfectly.

**Comment by shymega** [comment]
@Absolucy Could you merge `master` onto this PR, or at least, commit: a53380de4fcf62f9cd89a0e1398d0c6e0f44540f

The reason for that  is because CI has been updated to build on Big Sur, and I wanted to be sure your PR works. Cheers.

**Comment by Absolucy** [comment]
i don't even know how this PR is open anymore considering that my fork is deleted

**Comment by shymega** [comment]
Nor do I - do you still want it merged? I can checkout the PR and merge it myself if you want?

**Comment by Absolucy** [comment]
Yeah, feel free.

**Comment by shymega** [comment]
Right, OK, that's been merged. Thanks for the PR, its a welcome addition to our macOS support, both local builds and CI!

**Comment by fdelucchijr** [comment]
Thanks, @Absolucy!

**Pull request merged**